install/kubernetes/cilium/templates/cilium-operator-deployment.yaml

```diff
@@ -155,13 +155,30 @@ spec:
         - name: {{ $key }}
           value: {{ $value }}
 {{- end }}
+{{- $image := deepCopy .Values.operator.image -}}
 {{- if .Values.eni.enabled }}
-        image: {{ .Values.operator.image.repository }}-aws{{ .Values.operator.image.suffix }}:{{ .Values.operator.image.tag }}
+    {{- if get $image "repository" -}}
+        {{- $_ := set $image "repository" (printf "%s-%s" (get $image "repository") "aws") -}}
```

Comment:
Have you considered adding this logic to the `container.image` template? We could set something like

```
{{- $customSuffix := "-aws" }}
```

Here, and then invoke the template with:

```
{{ template "container.image" (dict "registry" $.Values.registry "image" $image "customSuffix" $customSuffix) }}
```

And the template itself does something like:

```
 {{- printf "%s%s%s:%s" .image.repository (.customSuffix | default "") (.image.suffix | default "") (required ".tag is required" .image.tag) | quote -}} 
```

Reply:
Agreed, would be nice to move `customSuffix` setting to a separate if-else block, seems like it would remove code duplication from here.

Reply:
Good idea, I thought about adding something like `customPrefix` to `image` but it felt weird to have a second prefix and push the burden at the `value.yaml` level. Your suggestion sounds great!